### ISSUE TITLE
implement hitTestPosition as in android version [new feature]

### ISSUE THREAD

**Issue by AbdullahRusi** [Tehreer/Tehreer-Cocoa#3]
السلام عليكم ورحمة الله وبركاته

I think this feature is nice to have in shaa Allah
to get tapped word with something like [this SO answer](https://stackoverflow.com/a/44984629)
or [that](https://stackoverflow.com/a/57629254)

this or implement shouldInteractWith like in standard textView
it's may be very useful when using custom URL schemes in app
as mentioned in [tutorial](https://www.hackingwithswift.com/example-code/system/how-to-make-tappable-links-in-nsattributedstring)

update: here is prepared xcode project with word-by-word flow layout with TLabel's
[WordByWord.zip](https://github.com/Tehreer/Tehreer-Cocoa/files/4001251/WordByWord.zip)

**Comment by mta452** [comment]
وعلکم السلام ورحمة اللہ وبرکاتة

Thanks for sharing the project and the links. After looking at the different approaches, I have decided that, for now, the simplest solution would be to align with the android version. So I have added the following methods in `TLabel`:
`indexOfCharacter(at: CGPoint) -> String.Index?` will return the character index in source string.
`indexOfCodeUnit(at: CGPoint) -> Int?` will return the convenient UTF-16 index that can be used with `NSString`.

Tested it with the following patch in the shared project.
```swift
private lazy var arWord: TLabel = {
    let label = TLabel()
    label.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(arWordTapped(_:))))

    return label
}()

@objc private func arWordTapped(_ sender: UITapGestureRecognizer) {
    let location = sender.location(in: arWord)
    if let index = arWord.indexOfCodeUnit(at: location) {
        print("Tapped Index: \(index)")
    }
}
```

Similarly, I'll add a `LinkableLabel` in demo project for the demonstration of URL handling.
For word boundary analysis, a wrapper over `libunibreak` will be provided in future version.
